File: app/src/main/java/cm/aptoide/pt/home/apps/model/StateApp.java
package cm.aptoide.pt.home.apps.model;

import cm.aptoide.pt.home.apps.App;

public interface StateApp extends App {

  boolean isIndeterminate();

  void setIndeterminate(boolean indeterminate);

  Status getStatus();

  void setStatus(Status status);

  int getProgress();

  enum Status {
    ACTIVE, STANDBY, COMPLETED, ERROR, UPDATE, UPDATING, PAUSING, INSTALLING
  }
}

File: old_app/src/main/java/cm/aptoide/pt/home/apps/model/StateApp.kt
package cm.aptoide.pt.home.apps.model

import cm.aptoide.pt.home.apps.App

interface StateApp : App {
  val md5: String
  val status: Status?
  val progress: Int
  val packageName: String
  val versionCode: Int

  enum class Status {
    ACTIVE, INSTALLING, PAUSE, ERROR, STANDBY, IN_QUEUE
  }
}